src/backend/common/half.hpp

```diff
@@ -919,10 +919,20 @@ AF_CONSTEXPR __DH__ static inline bool operator==(
     arrayfire::common::half lhs, arrayfire::common::half rhs) noexcept;
 AF_CONSTEXPR __DH__ static inline bool operator!=(
     arrayfire::common::half lhs, arrayfire::common::half rhs) noexcept;
+
+#ifdef AF_ONEAPI
+SYCL_EXTERNAL inline bool operator<(common::half lhs,
```

Comment:
I don't see the difference between this and the other branch. can you explain? if its because of SYCL_EXTERNAL it would be better to define it for the other backends rather duplicating the code here.

Reply:
external linkage required to call function from within a sycl kernel, also:
'sycl_device' attribute cannot be applied to a static function or function in an anonymous namespace